### PR TITLE
New page meta bar

### LINKED ISSUE #3829: Reorganize page meta bar
## Feature description

The Page meta bar has been changed in UX and might change even more.

First off, items have been reordered and undo and redo have been moved here (remove them from header):

Screen Shot 2019-11-27 at 10 35 06 AM

Secondly, it might completely move and be tied to the header rather than the page as shown here:

Screen Shot 2019-11-27 at 10 35 25 AM

Don't work on this until UX has finalized for design.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* Remove undo/redo from header bar.
* Add undo/redo to page bar.
* When UX finalizes: relayout page bar location and design to match UX.

## Implementation brief

* 

## QA testing instructions

* 

## Demo

* 

## Changelog entry

*

### PR DESCRIPTION
**Pull request by barklund**
## Summary

Implemented design for new page meta bar:
Screen Shot 2019-12-21 at 8 16 13 PM

Fixes #3829.

## Checklist

- [x] My pull request is addressing an [open issue](https://github.com/ampproject/amp-wp/contributing/project-management.md#life-of-an-issue) (please create one otherwise).
- [x] My code is tested and passes existing [tests](https://github.com/ampproject/amp-wp/contributing/engineering.md#tests).
- [x] My code follows the [Engineering Guidelines](https://github.com/ampproject/amp-wp/contributing/engineering.md) (updates are often made to the guidelines, check it out periodically).

### CODE REVIEW

**Comment by swissspidy** [comment]
Just now saw that there might be some upcoming design changes: https://github.com/ampproject/amp-wp/issues/3829#issuecomment-570741568

Still this PR sounds like a good start.

File: assets/src/edit-story/components/canvas/meta/index.js [hunk 1]
```diff
@@ -11,21 +11,69 @@ import { useCallback } from '@wordpress/element';
 /**
  * Internal dependencies
  */
-import { useStory } from '../../../app';
+import { PAGE_WIDTH } from '../../../constants';
+import { useStory, useHistory } from '../../../app';
 import Switch from '../../switch';
-import Delete from './delete.svg';
-import Duplicate from './duplicate.svg';
+import Delete from './delete_icon.svg';
```
Comment (spacedmonkey) [inline assets/src/edit-story/components/canvas/meta/index.js]: I think the icons should be removed to their own directory.

Comment (spacedmonkey) [reply assets/src/edit-story/components/canvas/meta/index.js]: @barklund ^ this hasn't been actions.

File: assets/src/edit-story/components/canvas/meta/index.js [hunk 2]
```diff
@@ -32,45 +55,104 @@ const Options = styled.div`
...
+					</Icon>
+					<Space />
+					<Divider />
+					<Space />
+					<Icon disabled={ ! canRedo } onClick={ handleRedo }>
+						<Redo />
+					</Icon>
+					<Space />
+					<Icon disabled={ ! canUndo } onClick={ handleUndo }>
+						<Undo />
```
Comment (swissspidy) [inline assets/src/edit-story/components/canvas/meta/index.js]: I've asked for clarification on Figma, but I assume the position of the undo and redo arrows should probably be switched here.

**Comment by spacedmonkey** [comment]
Screenshot 2020-01-15 at 11 19 55
I am unable to see this meta bar, is this the expected behaviour?

**Comment by swissspidy** [comment]
@spacedmonkey Perhaps need to scroll? I see it like this:

Screenshot 2020-01-15 at 12 29 22

**Comment by spacedmonkey** [comment]
> @spacedmonkey Perhaps need to scroll? I see it like this:
> 
@swissspidy  No I know that. But what I am saying is, why do I have to scroll on big screen like mine? It is weird.

File: assets/src/edit-story/components/canvas/canvasLayout.js [hunk 3]
```diff
@@ -23,9 +22,10 @@ const Background = styled.div`
 
 	display: grid;
 	grid:
-    ".   meta       .        ." 1fr
-    ".   page       addpage  ." ${ PAGE_HEIGHT }px
-    ".   carrousel  .        ." 1fr
+    ".     .          .        ." 1fr
```
Comment (swissspidy) [inline assets/src/edit-story/components/canvas/canvasLayout.js]: This space is now more like 36px.

**Comment by swissspidy** [comment]
True. It'll get better once the top bar and carousel have been changed according to the new design. This is just the first step. If it doesn't get better after those changes, we'll flag to UX.

File: assets/src/edit-story/components/canvas/canvasLayout.js (hunk 3 above)
Comment (swissspidy) [reply assets/src/edit-story/components/canvas/canvasLayout.js]:
```suggestion
    ".     .          .        ." 36px
```

**Comment by spacedmonkey** [comment]
> True. It'll get better once the top bar and carousel have been changed according to the new design. This is just the first step. If it doesn't get better after those changes, we'll flag to UX.

So my original quote

> is this the expected behaviour

So this is expected.

File: assets/src/edit-story/components/canvas/meta/index.js (hunk 1 above)
Comment (barklund) [reply assets/src/edit-story/components/canvas/meta/index.js]: As in a subdirectory of this directory or a single central directory? Also, why, if they're not reused?

**Comment by barklund** [comment]
> I am unable to see this meta bar, is this the expected behaviour? This is on a MacBook Pro 16"

For now, yes. There's a lot more changes coming to the overall structure (including #4110) and finally we have a ticket for responsive layout (#3920), which will make sure everything is stretching and collapsing correctly in the end.

File: assets/src/edit-story/components/canvas/canvasLayout.js (hunk 3 above)
Comment (barklund) [reply assets/src/edit-story/components/canvas/canvasLayout.js]: 36px? Where did you get that? It see 12/25 px between the top of the page canvas and the bottom of the publish button. Or 48/77 px between the top of the page canvas to the top of the editor overall. But I think this is best left for future responsive layout efforts post #4110.

File: assets/src/edit-story/components/canvas/meta/index.js (hunk 1 above)
Comment (swissspidy) [reply assets/src/edit-story/components/canvas/meta/index.js]: Subdirectory would make sense but it's not too urgent IMO. We can also do that for all the components in some dedicated cleanup/refactor PR.

**Comment by barklund** [comment]
@swissspidy, can we merge this? There's still some UX to be done about the page number (it might be moved), but we can deal with it then. Also, having this in before #4110 makes sense, I guess.

**Comment by swissspidy** [comment]
SGTM 👍

**Pull request merged**